Math/NumberTheory/Zeta/Utils.hs

```diff
@@ -0,0 +1,51 @@
+-- |
+-- Module:      Math.NumberTheory.Zeta.Utils
+-- Copyright:   (c) 2018 Alexandre Rodrigues Baldé
+-- Licence:     MIT
+-- Maintainer:  Alexandre Rodrigues Baldé <alexandrer_b@outlook.com>
+-- Stability:   Provisional
+-- Portability: Non-portable (GHC extensions)
+--
+-- Shared utilities used by functions from @Math.NumberTheory.Zeta@.
+
+module Math.NumberTheory.Zeta.Utils
+  ( intertwine
+  , skipEvens
+  , skipOdds
+  , suminf
+  ) where
+
+-- | Joins two lists element-by-element together into one, starting with the
+-- first one provided as argument.
+--
+-- >>> take 10 $ intertwine [0, 2 ..] [1, 3 ..]
+-- [0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
+intertwine :: [a] -> [a] -> [a]
+intertwine (x : xs) (y : ys) = x : y : intertwine xs ys
+intertwine      xs       ys  = xs ++ ys
```

Comment:
It does not matter for zeta and beta, but as long as this function may be used in other contexts... 
Such implementation of `intertwine` is unnecessary strict. E. g., `head $ intertwine [1] undefined` triggers evaluation of `undefined` instead of just returning `1`.

Reply:
@Bodigrim I do not see how to make it _not_ fail with that input.

Reply:
Hm, it'll work if it's a mutually recursive function, but isn't that overengineering?

Reply:
```haskell
intertwine :: [a] -> [a] -> [a]
intertwine [] ys = ys 
intertwine (x : xs) ys = x : intertwine ys xs 
```

Reply:
@Bodigrim passed me right by. Thanks.